test/integration/features/flags.feature

```diff
@@ -0,0 +1,82 @@
+@flags
+Feature: Flags
+  As a user I can perform advanced operations
+  such as starting Minishift with optional features
+
+  Scenario: Set configuration options for Minishift
+    Given Minishift has state "Does Not Exist"
+     When executing "minishift config set docker-opt dns=8.8.8.8" succeeds
+     Then executing "minishift config get docker-opt" succeeds
+      And stdout should contain
+      """
+      [dns=8.8.8.8]
+      """
+     When executing "minishift config set docker-env BAZ=BAT" succeeds
+     Then executing "minishift config get docker-env" succeeds
+      And stdout should contain
+      """
+      [BAZ=BAT]
+      """
+
+  Scenario: Starting Minishift
+    Given Minishift has state "Does Not Exist"
+     When executing "minishift start --insecure-registry test-registry:5000 --docker-env=FOO=BAR" succeeds
+     Then Minishift should have state "Running"
+
+  Scenario: User is able to get value of the insecure registry
+    Given Minishift has state "Running"
+     When executing "minishift ssh docker info" succeeds
+     Then stdout should contain
+      """
+      test-registry:5000
+      """
+
+  @b2d-only
+  Scenario: User is able to get value of the docker environment variables
+    Given Minishift has state "Running"
+     When executing "minishift ssh cat /var/lib/boot2docker/profile" succeeds
```

Comment:
> This means you would hide the actual ISO image information because somewhere we would still need to record the information about which image is used.

why? You would not have to. One way would be to get the release version of the ISO as part of the step implementation. This allows to differentiate between various ISOs. Depending on the release version, I can then make a decision on where to look.